### PR TITLE
Refactor that includes support for old API as of latest commit

### PR DESCRIPTION
**Pull request by ruslan120101**

### CODE REVIEW

**Comment by ruslan120101** [comment]
this branch is called "programmaticWithAnimation" but also fully supports storyboarding

File: MILRatingCollectionView/MILRatingCollectionView.swift [hunk 1]
```diff
@@ -6,406 +6,467 @@ Licensed Materials - Property of IBM
...
+        static var FontUnHighlightedAnimationScalingTransform: CGFloat { return (1 / FontHighlightedAnimationScalingTransform) }
+        
+    }
     
-    /// The background color of the circle that surrounds the selected item
-    let circleBackgroundColor = UIColor(red: 218.0/255.0, green: 87.0/255.0, blue: 68.0/255.0, alpha: 1.0)
     
-    var dummyBackgroundView: UIView!
-    var circularView: UIView!
-    var selectedIndexPath: NSIndexPath?
```
Comment (tfrank64) [inline MILRatingCollectionView/MILRatingCollectionView.swift]: Does this exist in some other form now? Or is there someway to know the starting cell?

File: MILRatingCollectionView/MILRatingCollectionView.swift [hunk 2]
```diff
@@ -6,406 +6,467 @@ Licensed Materials - Property of IBM
...
-        numberLabel.font = UIFont(name: "\(selectedFont)-Medium", size: 30)
-        self.contentView.addSubview(numberLabel)
+        cleanExistingViews()
+        createDummyOverlayView()
+        createCircularView()
+        addCircularViewToDummyOverlayView()
+        configureScrollViewExcludingContentSize()
+        configureScrollViewContentSizeAndPopulateScrollView()
+        configureInitialScrollViewHighlightedIndex()
+        animateCircleToCenter()
```
Comment (tfrank64) [inline MILRatingCollectionView/MILRatingCollectionView.swift]: Can we make `animateCircleToCenter` more of an option. Currently, not calling it causes the circle to remain mostly off screen.

File: MILRatingCollectionView/MILRatingCollectionView.swift [hunk 3]
```diff
@@ -6,406 +6,467 @@ Licensed Materials - Property of IBM
...
         
-        self.numberLabel.textColor = normalFontColor
-        self.numberLabel.font = UIFont(name: "\(selectedFont)-Medium", size: 30)
-        self.numberLabel.transform = CGAffineTransformScale(self.numberLabel.transform, 2.0, 2.0)
-        UIView.animateWithDuration(0.1, animations: {
-            self.numberLabel.transform = CGAffineTransformMakeScale(1.0,1.0)
-            
-        })
+        _dummyOverlayView = UIView(frame: _dummyViewFrame)
+        _dummyOverlayView.backgroundColor = UIColor.brownColor()
```
Comment (tfrank64) [inline MILRatingCollectionView/MILRatingCollectionView.swift]: The brown has got to go 😝

**Comment by tfrank64** [comment]
Overall, the code looks a lot cleaner and modular. Some comments and methods and computed properties would be great in addition to the comments I added to your changes.

File: MILRatingCollectionView/MILRatingCollectionView.swift (hunk 3 above)
Comment (ruslan120101) [inline MILRatingCollectionView/MILRatingCollectionView.swift]: :+1:

**Comment by ruslan120101** [comment]
thanks! and sounds good

File: MILRatingCollectionView/MILRatingCollectionView.swift (hunk 1 above)
Comment (ruslan120101) [inline MILRatingCollectionView/MILRatingCollectionView.swift]: **facepalm** fixed 

https://github.com/IBM-MIL/MILRatingCollectionView/commit/5edd45e4f5d4333cc7a6ba791bf01620dea241cf

**Comment by ruslan120101** [comment]
https://github.com/IBM-MIL/MILRatingCollectionView/commit/2401c68b9e4564268c9bd8636141a61481eb1190 

fixes the background color

**Comment by tfrank64** [comment]
Once the setter is added for selectedIndex, we should be good

**Comment by ruslan120101** [comment]
:' ]

**Pull request merged**